swift/ql/src/queries/Security/CWE-1204/StaticInitializationVector.ql

```diff
@@ -32,14 +32,15 @@ class StaticInitializationVectorSource extends Expr {
 class EncryptionInitializationSink extends Expr {
   EncryptionInitializationSink() {
     // `iv` arg in `init` is a sink
-    exists(CallExpr call, string fName |
+    exists(InitializerCallExpr call, string fName |
       call.getStaticTarget()
-          .(ConstructorDecl)
           .hasQualifiedName([
               "AES", "ChaCha20", "Blowfish", "Rabbit", "CBC", "CFB", "GCM", "OCB", "OFB", "PCBC",
               "CCM", "CTR"
             ], fName) and
-      call.getArgumentWithLabel("iv").getExpr() = this
+      call.getArgumentWithLabel("iv").getExpr() = this and
+      not call.isSelfCall() and
+      not call.isSuperCall()
```

Comment:
Correct, I added them because `OtherConstructorDeclRefExpr`s have now been largely subsumed under `MethodLookupExpr`. And in this query, where an initializer is a sink, and another `convenience` initializer does a `self.init` call to the one that is a sink, we want to exclude this call as a call to a sink.

But now that I think of it, maybe we do want to keep it as a sink after all, since we could be blocking legitimate paths that end up in the convenience initializer. I'll revisit this.

Reply:
Yeah, the problem was that both the normal init and the convenience init are sinks, and the convenience init calls into the normal init, so we don't want two separate paths for the convenience init (one that ends there and one that further goes into the other init). 

But on the other hand, this shouldn't be an issue in practice, since the source code of these library classes will not be visible to the extractor... Right?